std/container/binaryheap.d

```diff
@@ -125,11 +126,26 @@ if (isRandomAccessRange!(Store) || isRandomAccessRange!(typeof(Store.init[])))
     {
         assert(!store.empty, "Cannot pop an empty store.");
         if (store.length == 1) return;
-        auto t1 = store[].moveFront();
-        auto t2 = store[].moveBack();
+        static if(isRandomAccessRange!Store)
+        {
+            auto t1 = store.moveFront();
+            auto t2 = store.moveBack();      
+        }
+        else
+        {
+            auto t1 = store[].moveFront();
+            auto t2 = store[].moveBack();
+        }
```

Comment:
There's a lot of subtle duplication here and in the similar code that follows. Why is there a need to use `store` if the range is random access and `store[]` otherwise? Is it concerning a statically-sized array thing?

Anyhow, to fix this without duplication, put the conditional in a method:

```
auto storeRange()
{
    static if(isRandomAccessRange!Store)
        return store[];
    else
       return store;
}
```

Then use `storeRange` throughout (and get used to write calls without parens, regardless of all answers you'll see to this comment :o)).